packages/datadog-plugin-aws-sdk/src/services/kinesis.js

```diff
@@ -152,7 +152,12 @@ class Kinesis extends BaseAwsSdkPlugin {
       case 'putRecords':
         stream = params.StreamArn ? params.StreamArn : (params.StreamName ? params.StreamName : '')
         for (let i = 0; i < params.Records.length; i++) {
-          this.injectToMessage(span, params.Records[i], stream, i === 0)
+          this.injectToMessage(
+            span,
+            params.Records[i],
+            stream,
+            i === 0 || (this.config.kinesis && this.config.kinesis.batchPropagationEnabled)
```

Comment:
Can you hoist this just before the for loop?

```js
const enableKinesisBatch = this.config.kinesis && this.config.kinesis.batchPropagationEnabled
// ...
i === 0 || enableKinesisBatch
```

Reply:
done